### PR TITLE
Adding cisco nxos show hardware internal bigsur all-ports detail

### PR DESCRIPTION
**Pull request by nsnelson402**
ISSUE TYPE
New Template Pull Request

COMPONENT
cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm, cisco_nxos, show hardware internal bigsur all-ports detail

SUMMARY
Template to parse output of show hardware internal bigsur all-ports detail. Collect details for each port.

### CODE REVIEW

File: ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm [hunk 1]
```diff
@@ -0,0 +1,457 @@
+Value Required PORT_NAME (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm:1]:
```suggestion
Value Required PORT (\S+)
```

File: ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+Value Required IF_INDEX (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm:2]:
```suggestion
Value Required INTERFACE_INDEX (\S+)
```

File: ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+Value INSTANCE_SLOT (\S+)
+Value INSTANCE_ASIC (\S+)
+Value ASIC_EPORT (\S+)
+Value LOGICAL_PORT (\S+)
+Value FRONT_PORT (\S+)
+Value STATE ([\S\s]+?)
+Value BIGSUR_EPORT (\S+)
+Value PORT_TYPE (\S+)
+Value PORT_TYPE_DEC (\S+)
+Value TYPE (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm:13]: What is this for? Can we get a bit more?

File: ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+Value FW_INSTANCE (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm:14]:
```suggestion
Value FIREWALL_INSTANCE (\S+)
```

**Review by jvanderaa — COMMENTED** [review]
Would like to get a bit more explicit on the names, and align for others, such as PORT is part of the data model.

File: ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm (hunk 3 above)
Comment (nsnelson402) [reply ntc_templates/templates/cisco_nxos_show_hardware_internal_bigsur_all-ports_detail.textfsm:13]: I have no idea what the field bigsur eport is, but he supervisors have a code of 0 and the 40gbe have a code of 1
port_type indicates the string of the hardware type ie 40gbe
port_type_dec is the decimal code for the that type ie 12
I also don't know much about what the type field indicates other than that the sup and 40gbe differ.
I was just doing my best to make sure all of the variables provided in this output were accounted for so anyone else that wanted to use them could.

**Review by jvanderaa — APPROVED** [review]

**Pull request merged**